data-processing/common/commands/fetch_s2_data.py

```diff
@@ -27,7 +44,31 @@ class S2ReferencesNotFoundException(S2MetadataException):
     The target arxiv paper did not have any references available via the S2 public api.
     """
 
+
 class FetchS2Metadata(ArxivBatchCommand[ArxivId, S2Metadata]):
+    def __init__(self, args: Any) -> None:
+        super().__init__(args)
+        self._partner_api_token = os.getenv("S2_PARTNER_API_TOKEN", None)
+
+    def _mk_api_request(self, arxivId: ArxivId) -> requests.Response:
+        # XXX(andrewhead): S2 API does not have versions of arXiv papers. I don't think this
+        # will be an issue, but it's something to pay attention to.
+        versionless_id = self._strip_arxiv_version(arxivId)
+
+        if self._partner_api_token:
+            logger.info("Calling S2's partner API")
+            return requests.get(
```

Comment:
might consider not duplicating the request but rather have the fork change the url base + headers? can avoid the duplicate logging and multiple return statements